### PR TITLE
Add configuration for RedisTemplate

### LINKED ISSUE #2332: Add configuration for RedisTemplate
In the case of the PCF server, we plan to turn off Boot's RedisAutoConfiguration, so still need an explicit bean definition for RedisTemplate.

### PR DESCRIPTION
**Pull request by markpollack**
The code was copied out Boot's RedisAutoConfiguration.

Resolves #2332

### CODE REVIEW

**Comment by markpollack** [comment]
Thought this was needed to solve https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/issues/436  but it isn't.  closing

**Comment by sabbyanandan** [comment]
Looks like we need this after all. Without it, I see the following error.

```
2018-07-19T20:52:09.48-0700 [APP/PROC/WEB/0] OUT ***************************
   2018-07-19T20:52:09.48-0700 [APP/PROC/WEB/0] OUT APPLICATION FAILED TO START
   2018-07-19T20:52:09.48-0700 [APP/PROC/WEB/0] OUT ***************************
   2018-07-19T20:52:09.48-0700 [APP/PROC/WEB/0] OUT Description:
   2018-07-19T20:52:09.48-0700 [APP/PROC/WEB/0] OUT A component required a bean named 'redisTemplate' that could not be found.
   2018-07-19T20:52:09.48-0700 [APP/PROC/WEB/0] OUT Action:
```

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/AnalyticsConfiguration.java [hunk 1]
```diff
@@ -52,4 +56,24 @@ public FieldValueCounterRepository fieldValueCounterReader(RedisConnectionFactor
 	public AggregateCounterRepository aggregateCounterReader(RedisConnectionFactory redisConnectionFactory) {
 		return new RedisAggregateCounterRepository(redisConnectionFactory, new RetryTemplate());
 	}
+
+	@Bean
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/AnalyticsConfiguration.java]: I don't understand what these changes do as it looks to be exactly same as what boot auto-config would do?

Comment (jvalkeal) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/AnalyticsConfiguration.java]: Right, if only reason to reintroduce these here is to assume that CF server will disable redis auto-config, it kinda pointless. Then these should be added to CF server as just looking dataflow core these are really pointless here.

**Comment by sabbyanandan** [comment]
All the required changes are specific to CF-Server and it is addressed via spring-cloud/spring-cloud-dataflow-server-cloudfoundry#437. This is not required anymore.

**Pull request closed without merging**